mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAO.java

```diff
@@ -243,7 +244,15 @@ private PreparedStatement prepareSelectUidRangeLimited(Session session) {
         ComposedMessageId composedMessageId = metadata.getComposedMessageId().getComposedMessageId();
         Flags flags = metadata.getComposedMessageId().getFlags();
         ThreadId threadId = metadata.getComposedMessageId().getThreadId();
-        return cassandraAsyncExecutor.executeVoid(insert.bind()
+
+        BoundStatement boundStatement = insert.bind();
+        if (metadata.getComposedMessageId().getFlags().getUserFlags().length == 0) {
+            boundStatement.unset(USER_FLAGS);
```

Comment:
The if-else is necessary? 
When `length=0`, so it is equivalent to 
`boundStatement.setSet(USER_FLAGS, ImmutableSet.of());`, It still good?

Reply:
`boundStatement.setSet(USER_FLAGS, ImmutableSet.of());, It still good?`

No it is not: EMPTY collection DO insert tomstones....